### ISSUE TITLE
"hide ds4 controller" is not working

### ISSUE THREAD

**Issue by pablo5425** [Ryochan7/DS4Windows#2609]
thats it. even if its enabled, it wont hide controllers. this screenshot shows both dualsense and the emulated controller appearing in the list with the option enabled
![image](https://user-images.githubusercontent.com/34631489/192892319-4809efe9-3432-4e1c-a831-715622903bf0.png)
this causes double input in steam

edit: im not sure if this is part of the issue but gamepad tester shows that both triggers are being held in the original not emulated device
![image](https://user-images.githubusercontent.com/34631489/192893171-dccc9727-5dbe-4a70-8817-20b08080adf1.png)

**Comment by TheMasterDingo** [comment]
dont use that option, its very old, check hidhide

**Comment by pablo5425** [comment]
> dont use that option, its very old, check hidhide

i would if it had an easy way to enable and disable without doing too many clicks. bc i use ds4windows for external games and steam drivers for steam games where they can display ps buttons
basically the way i want to use it, hidhide must only be enabled if ds4windows is running

**Comment by pablo5425** [comment]
> > > dont use that option, its very old, check hidhide
> > 
> > 
> > i would if it had an easy way to enable and disable without doing too many clicks. bc i use ds4windows for external games and steam drivers for steam games where they can display ps buttons basically the way i want to use it, hidhide must only be enabled if ds4windows is running
> 
> Not sure if I understood correctly but you can add steam to the whitelist sonit always sees the native controller

adding steam makes it see both controllers. which is exactly what i want to avoid. it should only see the emulated controller

**Comment by TheMasterDingo** [comment]
Yes I understand now, the only thing would be to use a script to automate it, hidhide and ds4 support cmd line interface

**Comment by pablo5425** [comment]
i think ds4windows is taking slower to grab the controller. in fact on previous versions steam would never detect the controller if ds4windows was running even without hidhide

**Comment by pablo5425** [comment]
well, i found a workaround with the latest hidhide version
inverting cloak allows to switch between ds4w and steam drivers without causing conflicts without closing either process

**Comment by Yohoki** [comment]
I worked on the hidhide code for DS4Windows. We have an option in AutoProfiles to "Turn off DS4Windows Temporarily". But, since the use of HidHide is now recommended, that option just blocks all controller inputs, since DS4Windows isn't sending any and HidHide isn't sending any to the app.

I fixed this now, and any time you use the "Turn DS4Windows off temporarily" option, ds4windows will add the targeted game to hidhide's whitelist. So the new option is that DS4Windows will turn off, and your steam game should get physical controller inputs from HidHide.

TL:DR - Do this. It works automatically:
![image](https://user-images.githubusercontent.com/44595371/193920374-bb375f81-0ac9-4bd2-937c-cd4a2e88ac44.png)

**Comment by pablo5425** [comment]
> > > dont use that option, its very old, check hidhide
> > 
> > 
> > i would if it had an easy way to enable and disable without doing too many clicks. bc i use ds4windows for external games and steam drivers for steam games where they can display ps buttons basically the way i want to use it, hidhide must only be enabled if ds4windows is running
> 
> I worked on the hidhide code for DS4Windows. We have an option in AutoProfiles to "Turn off DS4Windows Temporarily". But, since the use of HidHide is now recommended, that option just blocks all controller inputs, since DS4Windows isn't sending any and HidHide isn't sending any to the app.
> 
> I fixed this now, and any time you use the "Turn DS4Windows off temporarily" option, ds4windows will add the targeted game to hidhide's whitelist. So the new option is that DS4Windows will turn off, and your steam game should get physical controller inputs from HidHide.
> 
> TL:DR - Do this. It works automatically: Install HidHide. Use "Turn off DS4Windows temporarily" option in an autoprofile. ![image](https://user-images.githubusercontent.com/44595371/193920374-bb375f81-0ac9-4bd2-937c-cd4a2e88ac44.png)

i want this setup - steam drivers on steam games. ds4windows in everything else
seems to be working by doing that method with steam.exe + steam games and i keep bluetooth on in ds4 settings but there is some problem with updating lightbar. it works perfectly when i switch to ds4windows mode but there is some small problem if i switch to steam mode

main light uses the right color after pressing guide button in any moment but player light wont turn on if i take too long pressing said button
this only happens if i alt+tab to already running steam games. opening a new window (i.e going into big picture or starting a game) does update all lights perfectly without manual work

**Comment by Yohoki** [comment]
> i want this setup - steam drivers on steam games. ds4windows in everything else seems to be working by doing that method with steam.exe + steam games and i keep bluetooth on in ds4 settings but there is some problem with updating lightbar. it works perfectly when i switch to ds4windows mode but there is some small problem if i switch to steam mode

...Did you TRY the suggestion?

Install HidHide. Add your steam games to DS4Windows under autoprofile. Tick the "Turn off DS4Windows" option for the steam games.

This will result in the situation you just described.

**Comment by pablo5425** [comment]
> > i want this setup - steam drivers on steam games. ds4windows in everything else seems to be working by doing that method with steam.exe + steam games and i keep bluetooth on in ds4 settings but there is some problem with updating lightbar. it works perfectly when i switch to ds4windows mode but there is some small problem if i switch to steam mode
> 
> Did you try the suggestion?
> 
> Install HidHide. Add your steam games to DS4Windows under autoprofile. Tick the "Turn off DS4Windows" option for the steam games.
> 
> This will result in the situation you just described.

i do have it like that
there is no double input or anything. just some small issues with lightbar with steam drivers trying to override lightbar settings

**Comment by Yohoki** [comment]
sweet! Glad that's working for you now.

As for the lightbar, that has nothing to do with DS4Windows, since it's now turned off in your game. I think I remember some one else mentioning it has to do with Windows. Something in windows turns the lightbar off at times, even though the controller is turned on. I'm afraid I don't know much more than that, though.

**Comment by Kanuan** [comment]
If using a DS4 version 1 controller (real via USB/BT, or via DS4Windows' virtual DS4v1) will suffer from this issue: Windows will turn off lightbar/reset rumble state everytime the user interacts with Windows'... windows/taskbar. It won't happen as long as the user stays inside of the game.

**Comment by pablo5425** [comment]
its a dualsense and it only happens when i alt+tab into a steam game. at least pressing ps button fixes it
id like to have it without that small detail but i can get used to press the button every time
